src/analyzer/analyzerqueue.cpp

```diff
@@ -1,468 +1,215 @@
+            kLogger.warning()
+                    << "Invalid number of worker threads:"
+                    << numWorkerThreads;
+    } else {
+        kLogger.debug()
+                << "Starting"
+                << numWorkerThreads
+                << "worker threads";
+    }
+    // 1st pass: Create worker threads
+    m_workers.reserve(numWorkerThreads);
+    for (int threadId = 0; threadId < numWorkerThreads; ++threadId) {
+        m_workers.emplace_back(std::make_unique<AnalyzerThread>(
+                threadId,
+                library->dbConnectionPool(),
+                pConfig));
+        connect(m_workers.back().thread(), SIGNAL(progress(int, AnalyzerThreadState, TrackId)),
+            this, SLOT(slotWorkerThreadProgress(int, AnalyzerThreadState, TrackId)));
+    }
+    // 2nd pass: Start worker threads
+    for (const auto& worker: m_workers) {
+        worker.thread()->start(kWorkerThreadPriority);
     }
-    return trackWaiting;
 }
 
-// This is called from the AnalyzerQueue thread
-// The returned track might be NULL, up to the caller to check.
-TrackPointer AnalyzerQueue::dequeueNextBlocking() {
-    QMutexLocker locked(&m_qm);
-    if (m_queuedTracks.isEmpty()) {
-        Event::end("AnalyzerQueue process");
-        m_qwait.wait(&m_qm);
-        Event::start("AnalyzerQueue process");
-
-        if (m_exit) {
-            return TrackPointer();
-        }
+void AnalyzerQueue::emitProgress() {
+    const auto now = Clock::now();
+    // If all enqueued tracks have been finished the signal is
+    // emitted independent of the time when the last signal has
+    // been emitted.
+    if (!allEnqueuedTracksFinished() &&
+            (now < (m_lastProgressEmittedAt + kProgressInhibitDuration))) {
+        // Don't emit progress update signal
+        return;
     }
+    m_lastProgressEmittedAt = now;
 
-    const PlayerInfo& info = PlayerInfo::instance();
-    TrackPointer pLoadTrack;
-    QMutableListIterator<TrackPointer> it(m_queuedTracks);
-    while (it.hasNext()) {
-        TrackPointer& pTrack = it.next();
-        DEBUG_ASSERT(pTrack);
-        // Prioritize tracks that are loaded.
-        if (info.isTrackLoaded(pTrack)) {
-            kLogger.debug() << "Prioritizing" << pTrack->getTitle() << pTrack->getLocation();
-            pLoadTrack = pTrack;
-            it.remove();
-            break;
+    int analyzerProgressSum = 0;
+    int analyzerProgressCount = 0;
+    for (const auto& worker: m_workers) {
+        if (worker.analyzerProgress() >= kAnalyzerProgressNone) {
+            analyzerProgressSum += worker.analyzerProgress();
+            ++analyzerProgressCount;
         }
     }
-
-    if (!pLoadTrack && !m_queuedTracks.isEmpty()) {
-        // no prioritized track found, use head track
-        pLoadTrack = m_queuedTracks.dequeue();
+    int analyzerProgressAvg;
+    if (analyzerProgressCount > 0) {
+        analyzerProgressAvg = analyzerProgressSum / analyzerProgressCount;
```

Comment:
Running 4, 6, or even 8 parallel percentage counters at high frequency clutters the ui, doesn't add much value and has a performance impact. My first solution was to adjust the finished count by doing a modulo calculation, which worked out really nice. But I'm not sure if this is still possible with the current design.

Reply:
What is wrong with this? 
Also, it is not high frequency, as we are limited at 10Hz now.
It is true that my implementation would still send a signal for each of the percentages, and maybe that could be improved (no quick idea right now of how to), but still I wouldn't consider it to have a detrimental effect.
(Ps: I have a small bug about the track counter and sometimes it shows one less, but since the branch is going to be closed, I don't care)

![image](https://user-images.githubusercontent.com/9913727/34539865-aab5d914-f0d2-11e7-9ae7-f8d577863a23.png)
![image](https://user-images.githubusercontent.com/9913727/34539885-b82752ee-f0d2-11e7-9e6d-b900af47cc1f.png)
![image](https://user-images.githubusercontent.com/9913727/34539895-bc7ef75c-f0d2-11e7-8847-179a02e99587.png)

Reply:
> What is wrong with this?

It shows an overwhelming amount of information to the user that they have no use for. I agree with @uklotzde, there should only be one percentage for the whole batch analysis.